lib/dom.d.ts

```diff
@@ -0,0 +1,4 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/// <reference lib="dom" />
```

Comment:
If we have dom-shim.d.ts, do we still need this lib reference directive? Or vice versa, given this do we need the shims? Feels like both is overkill.

Reply:
It looks that we need both
- dom-shim.d.ts only include a couple of the types that exposed in our public api and we ship it in the npm package.
- internally we use more DOM types, e.g., in xhr http client. dom.d.ts is for internal usage.

I need to make sure that we don't pack this file in the released package though.

Reply:
Gotcha! Thanks for explaining.

Reply:
@xirzec any other comments/questions, or this is good to merge?

Reply:
I'm happy!